jdaviz/configs/imviz/plugins/catalogs/catalogs.py

```diff
@@ -324,23 +328,58 @@ def plot_selected_points(self):
                                                            getattr(y, 'value', y))
 
     def vue_zoom_in(self, *args, **kwargs):
-        """This function will zoom into the image based on the selected points"""
+
+        self.zoom_to_selected(return_bounding_box=True)
+
+    def zoom_to_selected(self, padding=0.02, return_bounding_box=False):
+        """
+        Zoom to a region containing the currently selected points in the catalog.
+
+        Parameters
+        ----------
+        padding : float, optional
+            A fractional value representing the padding around the bounding box
+            of the selected points. It is applied as a proportion of the largest
+            dimension of the viewer's current limits. Defaults to 0.02.
+        return_bounding_box : bool, optional
+            If True, returns the bounding box of the zoomed region as
+            ((x_min, x_max), (y_min, y_max)). Defaults to False.
+
+        Returns
+        -------
+        tuple of tuple, optional
+            If there are activley selected rows, and `return_bounding_box` is
+            True, returns a tuple containing the bounding
+            box coordinates: ((x_min, x_max), (y_min, y_max)).
+            Otherwise, returns None.
+
+        """
+
+        viewer = self.app._jdaviz_helper._default_viewer
+
         selected_rows = self.table.selected_rows
 
+        if not len(selected_rows):
+            return
+
         x = [float(coord['x_coord']) for coord in selected_rows]
         y = [float(coord['y_coord']) for coord in selected_rows]
 
+        limits = viewer.state._get_reset_limits()
+        max_dim = max((limits[1] - limits[0]), (limits[3] - limits[2]))
+        padding = max_dim * padding
+
         # this works with single selected points
         # zooming when the range is too large is not performing correctly
-        x_min = min(x) - 50
-        x_max = max(x) + 50
-        y_min = min(y) - 50
-        y_max = max(y) + 50
+        x_min = min(x) - padding
```

Comment:
How did the delta change from 50 to 0.02?

Reply:
we discussed this in tagup this morning, i can explain offline

Reply:
in short: the units changed from pixels to fractional (see docstring)

Reply:
Ah ok. Should there be a check so `0 < padding <= 1`? Or at least `0 < padding`?

Reply:
i removed this test and replaced it with the updated value, since theres a check to keep 'padding' between 0 and 1.